api/envoy/admin/v2alpha/clusters.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+package envoy.admin.v2alpha;
+
+import "envoy/api/v2/core/base.proto";
+import "envoy/api/v2/core/address.proto";
+import "envoy/type/percent.proto";
+
+// [#protodoc-title: Clusters]
+
+// Admin endpoint uses this wrapper for `/clusters` to display cluster status information.
+// See :ref:`/clusters <operations_admin_interface_clusters>` for more information.
+message Clusters {
+  // Mapping from cluster name to each cluster's status.
+  repeated ClusterStatus cluster_statuses = 1;
+}
+
+// Details an individual cluster's current status.
+message ClusterStatus {
+  // Name of the cluster.
+  string name = 1;
+
+  // General outlier statistics if installed for this cluster.
+  OutlierInfo outlier_info = 2;
+
+  // Denotes whether this cluster was added via API or configured statically.
+  bool added_via_api = 3;
+
+  // Mapping from host address to the host's current status.
+  repeated HostStatus host_statuses = 4;
+}
+
+// :ref:`Cluster outlier detection <arch_overview_outlier_detection>` statistics. -1 for any of
+// these statistics denotes that there was not enough data to compute it.
+message OutlierInfo {
+  // The average success rate of the hosts in the Detector for the last aggregation interval.
+  //
+  // Note: the message will be omitted if there were not enough hosts with enough request volume to
+  // proceed with success rate based outlier ejection.
+  envoy.type.Percent success_rate_average = 1;
```

Comment:
Can this be derived from the individual hosts' `success_rate` by the consumer of the endpoint?

Reply:
Basically, yes (see below for a more detailed answer).  Is it worth providing it for usability?

They will be close.  Technically, they may differ under two scenarios IIUC:
1.  A host has been removed since the last time the success rates were updated (they operate on a timer).
2. I'm not 100% sure, but I think hosts that go from getting enough requests to be considered back to not having enough requests to be considered will report an outdated number for the `successRate()` call (this one is probably a bug since this goes against the documented behavior).  Same effect if the number of hosts drops back below the threshold for calculating the cluster-wide success rate average - none of their values get updated until the cluster gets back above that threshold.

Reply:
Unless the distinction is important to the use for (1) and (2), thenI would vote to just provide it at the host-level and allow the consumer to compute.

Reply:
Since the internally computed cluster-wide average is compared to each host's average to determine whether the host gets ejected or not, I would think the exact cluster-wide average might be important for users when debugging why a host got ejected.  However, it's something that's easy to add later if someone specifically requests it, so SGTM.